### PR TITLE
5630: Remove "Download releases" button from solutions with no release/distribution and from release page banner

### PR DESCRIPTION
**Pull request by claudiu-cristea**

### CODE REVIEW

File: web/modules/custom/asset_release/src/Controller/AssetReleaseController.php [hunk 1]
```diff
@@ -99,38 +104,27 @@ public function createAssetReleaseAccess(RdfInterface $rdf_entity, AccountInterf
...
-      ->execute();
-
-    /** @var \Drupal\rdf_entity\Entity\Rdf[] $releases */
-    $releases = Rdf::loadMultiple($ids);
-
-    // Filter out any release that the current user cannot access.
-    // @todo Filter out any unpublished release. See ISAICP-3393.
-    $releases = array_filter($releases, function ($release) {
-      return $release->access('view');
-    });
```
[the remark this answers is not in the document]
Comment (pfrenssen) [reply web/modules/custom/asset_release/src/Controller/AssetReleaseController.php]: Followup is https://webgate.ec.europa.eu/CITnet/jira/browse/ISAICP-5736

File: web/modules/custom/asset_distribution/asset_distribution.module [hunk 2]
```diff
@@ -251,3 +252,44 @@ function asset_distribution_form_alter(&$form, FormStateInterface $form_state, $
     }
   }
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_delete().
+ *
+ * Cleanup entity reference field items to the deleted distribution from
+ * solutions and releases.
```
[the remark this answers is not in the document]
Comment (claudiu-cristea) [reply web/modules/custom/asset_distribution/asset_distribution.module]: I know the performance issue by reading the core tickets but it's not affecting Joinup as a distribution CANNOT be referenced by more than one solution or release. So, only a parent entity reference is cleaned once.

File: web/modules/custom/asset_release/src/Controller/AssetReleaseController.php [hunk 3]
```diff
@@ -99,38 +104,27 @@ public function createAssetReleaseAccess(RdfInterface $rdf_entity, AccountInterf
...
    *   The solution rdf entity.
    *
-   * @return array
-   *   The build array for the page.
+   * @return array|\Drupal\Core\Routing\TrustedRedirectResponse
+   *   The build array for the page or a 404 response.
    */
   public function overview(RdfInterface $rdf_entity) {
-    // Retrieve all releases for this solution.
-    $ids = $this->queryFactory->get('rdf_entity')
```
[the remark this answers is not in the document]
Comment (claudiu-cristea) [reply web/modules/custom/asset_release/src/Controller/AssetReleaseController.php]: Nice catch!

File: web/modules/custom/solution/solution.module [hunk 4]
```diff
@@ -213,26 +227,54 @@ function solution_entity_extra_field_info(): array {
...
+            'mdl-js-ripple-effect',
+            'mdl-button--accent',
+          ],
+        ],
+        '#cache' => [
+          // Vary by route so that the 'group_header_block' will have different
+          // versions for 'asset_release.solution_asset_release.overview' and
+          // the canonical solution routes.
+          'contexts' => ['route'],
+        ],
```
[the remark this answers is not in the document]
Comment (claudiu-cristea) [reply web/modules/custom/solution/solution.module]: Added a dedicated cache context `route.name.is_download_releases_route`.

**Pull request merged**